pandas/core/arrays/period.py

```diff
@@ -539,21 +796,230 @@ def _check_timedeltalike_freq_compat(self, other):
                                     .format(cls=type(self).__name__,
                                             freqstr=self.freqstr))
 
+    def repeat(self, repeats, *args, **kwargs):
+        """
+        Repeat elements of a Categorical.
 
-PeriodArrayMixin._add_comparison_ops()
-PeriodArrayMixin._add_datetimelike_methods()
+        See also
+        --------
+        numpy.ndarray.repeat
+        """
+        # TODO(DatetimeArray): remove
+        nv.validate_repeat(args, kwargs)
+        values = self._data.repeat(repeats)
+        return type(self)(values, self.freq)
+
+    # Delegation...
+    def strftime(self, date_format):
+        return self._format_native_types(date_format=date_format)
+
+    def astype(self, dtype, copy=True):
+        # TODO: Figure out something better here...
+        # We have DatetimeLikeArrayMixin ->
+        #     super(...), which ends up being... DatetimeIndexOpsMixin?
+        # this is complicated.
+        # need a pandas_astype(arr, dtype).
+        from pandas import Categorical
+
+        dtype = pandas_dtype(dtype)
+
+        if is_object_dtype(dtype):
+            return np.asarray(self, dtype=object)
+        elif is_string_dtype(dtype) and not is_categorical_dtype(dtype):
+            return self._format_native_types()
+        elif is_integer_dtype(dtype):
+            values = self._data
+
+            if values.dtype != dtype:
+                # int32 vs. int64
+                values = values.astype(dtype)
+
+            elif copy:
+                values = values.copy()
+
+            return values
+        elif (is_datetime_or_timedelta_dtype(dtype) and
+              not is_dtype_equal(self.dtype, dtype)) or is_float_dtype(dtype):
+            # disallow conversion between datetime/timedelta,
+            # and conversions for any datetimelike to float
+            msg = 'Cannot cast {name} to dtype {dtype}'
+            raise TypeError(msg.format(name=type(self).__name__, dtype=dtype))
+        elif is_categorical_dtype(dtype):
+            return Categorical(self, dtype=dtype)
+        elif is_period_dtype(dtype):
+            return self.asfreq(dtype.freq)
+        else:
+            return np.asarray(self, dtype=dtype)
+
+    def _item(self):
```

Comment:
what is this used for?

Reply:
To support `PeriodIndex.item`, while avoiding unnecessary methods on PeriodArray.

But I see now it would be clearer to just implement this logic in `PeriodIndex itself.